include/seqan/index/find2_index_approx.h

```diff
@@ -0,0 +1,585 @@
+                                  uint8_t const errors,
+                                  Search<nbrBlocks> const & s,
+                                  uint8_t const blockIndex,
+                                  uint8_t const minErrorsLeftInBlock,
+                                  TDir const & /**/,
+                                  TDistanceTag const & /**/)
+{
+    bool goToRight = std::is_same<TDir, Rev>::value;
+    if (goDown(iter, TDir()))
+    {
+        uint32_t charsLeft = s.blocklength[blockIndex] - (needleRightIt - needleLeftIt - 1);
+        do
+        {
+            bool delta = !ordEqual(parentEdgeLabel(iter, TDir()),
+                                   needle[goToRight ? needleRightIt - 1 : needleLeftIt - 1]);
+
+            // NOTE (cpockrandt): this might not be optimal yet! we have more edges than in the theoretical model,
+            // since we go down an edge before we check whether it can even work out!
+            if (!std::is_same<TDistanceTag, EditDistance>::value && minErrorsLeftInBlock > 0 &&
+                charsLeft - 1 < minErrorsLeftInBlock - delta)
+                continue;
+
+            int32_t needleLeftIt2 = needleLeftIt - !goToRight;
+            uint32_t needleRightIt2 = needleRightIt + goToRight;
+
+            if (needleRightIt - needleLeftIt == s.blocklength[blockIndex])
+            {
+                // leave the possibility for one or multiple deletions! therefore, don't change direction, etc!
+                if (std::is_same<TDistanceTag, EditDistance>::value)
+                {
+                    _schemeSearchDeletion(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + delta, s,
+                                          blockIndex, TDir());
+                }
+                else
+                {
+                    uint8_t blockIndex2 = std::min(blockIndex + 1, static_cast<uint8_t>(s.u.size()) - 1);
+                    bool goToRight2 = s.pi[blockIndex2] > s.pi[blockIndex2 - 1];
+                    if (goToRight2)
+                        _schemeSearch(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + delta, s,
+                                      blockIndex2, Rev(), TDistanceTag());
+                    else
+                        _schemeSearch(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + delta, s,
+                                      blockIndex2, Fwd(), TDistanceTag());
+                }
+            }
+            else
+            {
+                _schemeSearch(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + delta, s, blockIndex,
+                              TDir(), TDistanceTag());
+            }
+
+            // Deletion
+            if (std::is_same<TDistanceTag, EditDistance>::value)
+                _schemeSearch(delegate, iter, needle, needleLeftIt, needleRightIt, errors + 1, s, blockIndex, TDir(),
+                              TDistanceTag());
```

Comment:
use braces here, too, otherwise it looks like _schemeSearch is part of the condition and the Tag is the action!